File: backend/src/main/java/com/pratik/hotelreservation/service/HotelService.java
package com.pratik.hotelreservation.service;

import com.pratik.hotelreservation.dto.request.HotelCreateRequest;
import com.pratik.hotelreservation.dto.request.HotelUpdateRequest;
import com.pratik.hotelreservation.dto.response.HotelResponse;

import java.util.List;

public interface HotelService {

    HotelResponse createHotel(HotelCreateRequest request);

    HotelResponse updateHotel(Long id, HotelUpdateRequest request);

    HotelResponse getHotelById(Long id);

    List<HotelResponse> getAllHotels();

    void deleteHotel(Long id);
}

File: frontend/src/services/hotelService.js
import api from "./api";

export const getAllHotels = async () => {
  const response = await api.get("/hotels");

  return response.data;
};

export const getHotelById = async (hotelId) => {
  const response = await api.get(`/hotels/${hotelId}`);

  return response.data;
};

export const createHotel = async (hotelData) => {
  const response = await api.post(
    "/hotels",
    hotelData
  );

  return response.data;
};

export const updateHotel = async (hotelId, hotelData) => {
  const response = await api.put(
    `/hotels/${hotelId}`,
    hotelData
  );

  return response.data;
};

export const deleteHotel = async (hotelId) => {
  const response = await api.delete(`/hotels/${hotelId}`);

  return response.data;
};
